src/containers/ProfileDetail/RoleFilter/RoleFilter.tsx

```diff
@@ -0,0 +1,55 @@
+import { MouseEvent } from 'react';
+import { RoleType } from '@interfaces/studyList';
+
+import { Background, Button } from './Role.style';
+
+interface Props {
+  role: RoleType;
+  onRoleChange: (role: RoleType) => void;
+}
+
+function RoleFilter({ role, onRoleChange }: Props) {
+  const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
+    const value = e.currentTarget.dataset.value as
```

Comment:
이부분은 선택적인 부분이긴 한데 currentTarget에 dataset이 없는 경우와 dataset에 value가 없는 경우가 있어
```suggestion
    const value = e.currentTarget?.dataset?.value
```
위와 같이 바꾸거나 
```suggestion
    const {dataset={value:''}} = e.currentTarget
    const value = dataset.value
```
위와 같은 식으로 기본값을 지정해주면 더욱 안정적인 코드가 될것 같습니다!!